### ISSUE TITLE
Linux build process not working as expected

### ISSUE THREAD

**Issue by ntfwc** [ZQuestClassic/ZQuestClassic#117]
Following the steps as written in the readme:
```
cd build
cmake ..
make
```
`make` fails with `fatal error: allegro/platform/alunixac.h: No such file or directory`

Copying alunixac.h from allegro/extra to allegro/include/allegro/platform allows it to build. But the resulting zelda binary does not have sound, and looking at it with ldd reveals that, for some reason, libasound is not linked.

On the 2.50.2 build from the website, sound does work and it has libasound linked.

I did try getting a fresh copy of allegro 4.4.2 to generate an alunixac.h. The result did define a few alsa variables the other did not. But copying that to allegro/include/allegro/platform still gave me the same result.

Linux Version: Xubuntu 16.04 64-bit
Commit: c8701fc9320a873e9542483c11398272ac0ccdad

**Comment by evouga** [comment]
How did you compile Allegro? (You may have to tinker with the configuration to enable sound support? You can try asking at the Allegro forums.) Does sound work in any application using that build of Allegro?

**Comment by evouga** [comment]
You can also try compiling the various sound libraries from source (in the other/ directory) rather than using the provided binaries.

Sorry I can't help more; the only Linux machine I use has never been able to get sound to work with ZC. The developer who compiled the 2.50.2 binaries is no longer actively involved with development, and the rest of us use Windows.

**Comment by ntfwc** [comment]
That's unfortunate. To clarify, I didn't compile allegro, I just let ZC use the pre-compiled library from the git repo. I can try building it and make another attempt.

**Comment by ntfwc** [comment]
So I looked at it over the weekend. I did succeed in building allegro with alsa, but I ended up figuring out that I don't actually need alsa for the audio to work.

It turned my sound problem stemmed from the fact that I was depending on it to pipe audio through pulseaudio, which, without alsa, is not what is done. Sound does work if I run it from my main desktop user account, which, from what I gather, means it has to be run from an account with direct access to the sound device. So it shouldn't be a problem in most cases.

Everything but midi music is working, which I investigated and I'll add a different issue for.

I have added a pull request with a fix for alunixac.h
